PlayerStrategy/AggressivePlayerStrategy.cpp

```diff
@@ -2,15 +2,239 @@
+
+		cout << "\nEnding turn\n" << endl;
+		return false;
+	}
+	
+	else {
+		
+		if(hand->getHandSize() > 0){
+
+			Card* card = hand->getCards().back();
+			Territory* target = map->getTerritory(toAttackTerritories[rand() % toAttackTerritoriesSize].getName());
+
+			if(card->getType() == Card::ReinforcementCT){
+
+				player->setTroopsToDeploy(*player->getTroopsToDeploy() + 5);
+				cout << "\nAdding 5 troops to deploy";
+			}
+
+			else if(card->getType() == Card::BombCT){
+
+				
+				ordersList->add(new Bomb(player->getPlayerID(), "", target));
+				cout << "\nBombing " << target->getName();
+			}
+
+			else if(card->getType() == Card::AirliftCT){
+
+				Territory* secondStrongest = getSecondStrongestTerritory(player);
+
+				ordersList->add(new Airlift(player->getPlayerID(), "", secondStrongest->getArmy(), secondStrongest, strongestTerritory));
+				cout << "\nAirlifting " << secondStrongest->getArmy() << " troops from " << secondStrongest->getName() << " to " << strongestTerritory->getName();
+			}
+
+			else{
+				cout << "\nNot issuing orders";
+			}
+
+			deck->returnCard(card);
+			hand->removeCard(card);
+
+			cout << "\nEnding turn\n" << endl;
+			return false;
+		}
+
+		else{
+
+			bool alreadyAdvancing = false;
+			for(Territory* neighbor : strongestTerritory->getNeighbors()){
+
+				for(Order* order : player->getOrdersList()->getOrders()){
+
+					if(dynamic_cast<Advance*>(order) != nullptr && dynamic_cast<Advance*>(order)->getSource() == neighbor){
+						alreadyAdvancing = true;
+					}
+				}
+
+				if(neighbor->getPlayer() == player->getPlayerID() && neighbor->getArmy() > 0 && !alreadyAdvancing){
+
+					ordersList->add(new Advance(player->getPlayerID(), "", neighbor->getArmy(), neighbor, strongestTerritory));
+					cout << "\nAdvancing " << neighbor->getArmy() << " troops from " << neighbor->getName() << " to " << strongestTerritory->getName();
+					cout << "\nEnding turn\n" << endl;
+					return false;
+				}
+			}
+
+			Territory* target = map->getTerritory(toAttackTerritories[rand() % toAttackTerritoriesSize].getName());
+			
+			ordersList->add(new Advance(player->getPlayerID(), "", std::numeric_limits<int>::max(), strongestTerritory, target));
+			cout << "\nAdvancing all troops from " << strongestTerritory->getName() << " to " << target->getName();
+			cout << "\nEnding turn\n" << endl;
+			return true;
+		}
+
+		cout << "\nEnding turn\n" << endl;
+		return true;
+	}
 }
 
 Territory* AggressivePlayerStrategy::toAttack(Player* player) {
-	return NULL;
+
+	Territory* strongestTerritory = getStrongestTerritory(player);
+
+	int count = 0;
+
+	for(Territory* territory : strongestTerritory->getNeighbors()){
+
+		if(territory->getPlayer() != player->getPlayerID()){
+			count++;
+		}
+	}
+
+	Territory* enemies = new Territory[count];
+
+	int i = 0;
+
+	for(Territory* territory : strongestTerritory->getNeighbors()){
+
+		if(territory->getPlayer() != player->getPlayerID()){
+			enemies[i] = *territory;
+			i++;
+		}
+	}
+
```

Comment:
player->setSizeOfToAttack(count);
    return toAttack;
    
    i believe this could replace the entire function  :int AggressivePlayerStrategy::toAttackSize(Player* player){...}

Reply:
Implemented and remove the method in new commit.